nixos/modules/services/home-automation/zwave-js.nix

```diff
@@ -0,0 +1,150 @@
+{config, pkgs, lib, ...}:
+
+with lib;
+
+let
+  cfg = config.services.zwave-js;
+  cacheDir = "/var/cache/zwave-js";
+  runtimeDir = "/run/zwave-js";
+  defaultConfig = {
+    storage = {
+      inherit cacheDir;
+    };
+  };
+  mergedConfigFile = "${runtimeDir}/config.json";
+in {
+
+  options.services.zwave-js = {
+    enable = mkEnableOption (mdDoc "Start the zwave-js server on boot");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.zwave-js-server;
+      defaultText = literalExpression "pkgs.zwave-js-server";
+      description = mdDoc ''
+        Package for zwave-js server
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 3000;
+      description = mdDoc ''
+        Port for the server to listen on.
+      '';
+    };
+
+    serialPort = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Serial port device path for Z-Wave controller.
+      '';
+      example = "/dev/ttyUSB0";
+    };
+
+    secretsConfigFile = mkOption {
+      type = types.path;
+      description = mdDoc ''
+        JSON file containing secret keys. This file will be merged with the
+        module-generated config file (taking precedence).
+
+        Z-Wave keys can be generated with:
+
+          {command}`< /dev/urandom tr -dc A-F0-9 | head -c32 ;echo`
+
+        ::: {.warning}
+        A file in the nix store should not be used since it will be readable to
+        all users.
+        :::
```

Comment:
Is there any upstream documentation we can reference, to what secret keys should be generated?

Reply:
I've added a link to the detailed upstream documentation of security keys and an example.